### PR TITLE
feat(dashboardeditor): send card select event

### PR DESCRIPTION
**Pull request by scottdickerson**
Closes #

**Summary**

- Adds a new onCardSelect event to the dashboard editor and fixes a few style issues with the dashboardeditor cards

**Change List (commits, features, bugs, etc)**

- fix(DataSeriesFormContent,TableSeriesFormContent): forced to regen the Combobox if the items change
- feat(DashboardEditor): call the onCardSelect callback when the card is selected (in every case)
- fix(dashboardEditor.scss): fixes the focus outline to not be inside the card
- fix(tableCard): fixes the table title font size and padding to match the other cards

**Acceptance Test (how to verify the PR)**

- Verify that as you select cards in the first dashboard story onCardSelect is triggered
- Verify that the table card focus state works when the Card is in the DashboardEditor, and that the title font-size and padding is correct

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Verify that the normal table title font sizes/paddings don't change

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by bjornalm — CHANGES_REQUESTED** [review]
Card selection looks good, but there is now a black border next to the blue border when resizing the cards.

File: packages/react/src/components/TableCard/_table-card.scss [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
-.#{$iot-prefix}--table-card .#{$iot-prefix}--card--content {
-  padding: 0;
+.#{$iot-prefix}--table-card {
+  .#{$iot-prefix}--card--content {
+    padding: 0;
+  }
+  .#{$iot-prefix}--table-toolbar-secondary-title {
+    // title needs to match new card title font size and padding
+    font-size: 1rem;
+    padding-top: 0.75rem;
```
[the remark this answers is not in the document]
Comment (scottdickerson) [reply packages/react/src/components/TableCard/_table-card.scss:10]: thanks for catching both of these, refixing and pushing

File: packages/react/src/components/TableCard/_table-card.scss [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 
-.#{$iot-prefix}--table-card .#{$iot-prefix}--card--content {
-  padding: 0;
+.#{$iot-prefix}--table-card {
+  .#{$iot-prefix}--card--content {
+    padding: 0;
+  }
+  .#{$iot-prefix}--table-toolbar-secondary-title {
+    // title needs to match new card title font size and padding
+    font-size: 1rem;
```
[the remark this answers is not in the document]
Comment (scottdickerson) [reply packages/react/src/components/TableCard/_table-card.scss:9]: thanks for catching this @bjornalm refixing and resubmitting

**Review by bjornalm — APPROVED** [review]
Nice work!

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/TableCardFormItems/TableCardFormContent.jsx [hunk 3]
```diff
@@ -331,7 +332,9 @@ const TableCardFormContent = ({
       >
         <ComboBox
           // need to re-gen if selected card changes or if a dataItem is removed from the list
-          key={`data-item-select-${removedDataItems.length}-selected_card-id-${cardConfig.id}`}
+          key={`data-item-select-${hash(validDataItemsForDropdown || {})}-selected_card-id-${
```
Comment (davidicus) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/TableCardFormItems/TableCardFormContent.jsx:335]: Are we trying to use key to re-render if validDataItemsForDropdown changes? Do we have to hash this or can we just pass in the obj?

**Review by davidicus — COMMENTED** [review]
It looks good. I just have the one question about the new object-hash package

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/TableCardFormItems/TableCardFormContent.jsx (hunk 3 above)
Comment (davidicus) [reply packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/TableCardFormItems/TableCardFormContent.jsx:335]: can perhaps be replaced by `(() => obj)()`

**Review by davidicus — APPROVED** [review]

**Comment by davidicus** [comment]
Could we possibly write up the values of using this hash package for keys at some time? I think it would be good to document so we can be consistent where possible.

**Pull request merged**